libs/core/langchain_core/prompts/base.py

```diff
@@ -82,9 +82,14 @@ def get_input_schema(
         self, config: Optional[RunnableConfig] = None
     ) -> Type[BaseModel]:
         # This is correct, but pydantic typings/mypy don't think so.
+        required_input_variables = {
+            k: (self.input_types.get(k, str), ...) for k in self.input_variables
+        }
+        optional_input_variables = {
+            k: (self.input_types.get(k, str), None) for k in self.partial_variables
+        }
         return create_model(  # type: ignore[call-overload]
-            "PromptInput",
-            **{k: (self.input_types.get(k, str), None) for k in self.input_variables},
+            "PromptInput", **{**required_input_variables, **optional_input_variables}
```

Comment:
partial_variables are variables that have already been provided. So partial variables should not appear as part of the input schema -- they're unrelated to optional and required variables

Reply:
I am going to add a new attribute to represent the optional variables.